cypress/e2e/CopyStateLinkButton.spec.ts

```diff
@@ -0,0 +1,152 @@
+        ).as('fetchTimeseriesCountData');
+        cy.intercept(
+            'GET',
+            'https://monkeypox-aggregates.s3.eu-central-1.amazonaws.com/total/latest.json',
+            { fixture: 'totalCasesData.json', statusCode: 200 },
+        ).as('fetchTotalCasesData');
+
+        cy.visit('/');
+
+        cy.wait(
+            [
+                '@fetchCountriesData',
+                '@fetchTotalCasesData',
+                '@fetchTimeseriesData',
+                '@fetchTimeseriesCountData',
+            ],
+            { timeout: 15000 },
+        );
+    });
+
+    it('Displays button', () => {
+        cy.visit('/');
+        cy.url().should('include', '/country');
+
+        cy.get('.popup-small-screens').should('not.exist');
+        cy.contains(/COPY LINK TO/i).should('be.visible');
+
+        cy.visit('/chart');
+        cy.get('.popup-small-screens').should('not.exist');
+        cy.contains(/COPY LINK TO/i).should('be.visible');
+    });
+
+    it('Copying proper link to clipboard', () => {
+        cy.visit('/country');
+        cy.wrap(
+            Cypress.automation('remote:debugger:protocol', {
+                command: 'Browser.grantPermissions',
+                params: {
+                    permissions: [
+                        'clipboardReadWrite',
+                        'clipboardSanitizedWrite',
+                    ],
+                    origin: window.location.origin,
+                },
+            }),
+        );
+
+        cy.wait(3000);
+
+        cy.contains('Argentina').click();
+
+        cy.get('.MuiSlider-track + span > input').invoke('attr', 'value', 12);
+
+        cy.contains(/COPY LINK TO/i)
+            .should('be.visible')
+            .focus()
+            .click();
+
+        cy.window().then((win) => {
+            win.navigator.clipboard.readText().then((text) => {
+                expect(text).to.include('?name=ARG&currDate=12');
+            });
+        });
+
+        cy.visit('/chart');
+
+        cy.wait(3000);
+
+        cy.contains('Argentina').click();
+
+        //changing slider via attributes, cases in test = 12, 25% of 12 is 3, 9 - 3 = 6, 50% of 12 is 6
+        cy.get('.MuiSlider-track').invoke('attr', 'style', [
+            'left: 25%',
+            'width: 50%',
+        ]);
+
+        cy.contains(/COPY LINK TO/i)
+            .should('be.visible')
+            .focus()
+            .click();
+
+        cy.window().then((win) => {
+            win.navigator.clipboard.readText().then((text) => {
+                expect(text).to.include('?name=ARG&startDate=3&endDate=9');
+            });
+        });
+    });
+
+    it('Display Alert', () => {
+        cy.visit('/country?name=test&currDate=test');
+
+        cy.wait(2000);
+        cy.get('.MuiSlider-track').should(
+            'have.attr',
+            'style',
+            'left: 0%; width: 100%;',
+        );
+        cy.get('.MuiAlert-message').should('exist');
```

Comment:
It's a good practice whenever it's possible to test the behaviour of the application from the user perspective. So here for example you can check whether text that should be displayed in the alert is visible, rather than checking if the class exists.